File: 388/388_version1.py
class Solution:
	def lengthLongestPath(self, input: str) -> int:
        files = input.split("\n")
        stack = [0]*(len(files)+1)
        level_sp = "\t"
        file_ex = "."
        max_len = 0
        print(files)
        for f in files:
            level = f.count("\t")
            stack[level+1] = stack[level]+len(f)-level+1
            if file_ex in f:
                max_len = max([stack[level+1]-1,max_len])
        return max_len

File: 388/388_version1.java
class Solution{
	public int lengthLongestPath(String input) {
        	String[] files = input.split("\n");
        	int[] stack = new int[files.length+1];
        	int maxLen = 0;
        	for(String f:files){
            		int level = f.lastIndexOf("\t")+1;
           		stack[level+1] = stack[level]+f.length()-level+1;
            		if(f.contains(".")) maxLen = Math.max(stack[level+1]-1,maxLen);
        }
        return maxLen;
    }
}
